core/settings/qPrefUnit.h

```diff
@@ -29,40 +29,53 @@ class qPrefUnits : public QObject {
 
 public:
 	static bool coordinates_traditional() { return prefs.coordinates_traditional; }
-	static units::DURATION duration_units() { return prefs.units.duration_units; }
-	static units::LENGTH length() { return prefs.units.length; }
-	static units::PRESSURE pressure() { return prefs.units.pressure; }
+	static QString duration_units();
+	static QString length();
+	static QString pressure();
 	static bool show_units_table() { return prefs.units.show_units_table; }
-	static units::TEMPERATURE temperature() { return prefs.units.temperature; }
+	static QString temperature();
 	static QString unit_system();
-	static units::TIME vertical_speed_time() { return prefs.units.vertical_speed_time; }
-	static units::VOLUME volume() { return prefs.units.volume; }
-	static units::WEIGHT weight() { return prefs.units.weight; }
+	static QString vertical_speed_time();
+	static QString volume();
+	static QString weight();
 
 public slots:
 	static void set_coordinates_traditional(bool value);
 	static void set_duration_units(units::DURATION value);
-	static void set_length(units::LENGTH value);
-	static void set_pressure(units::PRESSURE value);
+	static void set_duration_units(const QString& value);
+    static void set_length(units::LENGTH value);
+	static void set_length(const QString& value);
+    static void set_pressure(units::PRESSURE value);
+	static void set_pressure(const QString& value);
```

Comment:
at least from the way this looks on GitHub, this appears to introduce white space issues

Reply:
corrected, sorry for the problem.